src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/model/translator.kt

```diff
@@ -0,0 +1,139 @@
+@file:Suppress("ktlint:filename")
+
+package uk.gov.justice.digital.hmpps.prisonersearchindexer.model
+
+import uk.gov.justice.digital.hmpps.prisonersearchindexer.services.IncentiveLevel
+import uk.gov.justice.digital.hmpps.prisonersearchindexer.services.RestrictivePatient
+import uk.gov.justice.digital.hmpps.prisonersearchindexer.services.canonicalPNCNumberLong
+import uk.gov.justice.digital.hmpps.prisonersearchindexer.services.canonicalPNCNumberShort
+import uk.gov.justice.digital.hmpps.prisonersearchindexer.services.dto.nomis.OffenderBooking
+
+fun PrisonerA(ob: OffenderBooking, incentiveLevel: IncentiveLevel?, restrictedPatientData: RestrictivePatient?) =
+  PrisonerA().apply { this.translate(null, ob, Result.success(incentiveLevel), restrictedPatientData) }
+fun PrisonerA(existingPrisoner: Prisoner?, ob: OffenderBooking, incentiveLevel: Result<IncentiveLevel?>, restrictedPatientData: RestrictivePatient?) =
+  PrisonerA().apply { this.translate(existingPrisoner, ob, incentiveLevel, restrictedPatientData) }
+
+fun PrisonerB(ob: OffenderBooking, incentiveLevel: IncentiveLevel?, restrictedPatientData: RestrictivePatient?) =
+  PrisonerB().apply { this.translate(null, ob, Result.success(incentiveLevel), restrictedPatientData) }
+fun PrisonerB(existingPrisoner: Prisoner?, ob: OffenderBooking, incentiveLevel: Result<IncentiveLevel?>, restrictedPatientData: RestrictivePatient?) =
+  PrisonerB().apply { this.translate(existingPrisoner, ob, incentiveLevel, restrictedPatientData) }
```

Comment:
I think we want to move away from this PrisonerA / PrisonerB to represent prisoners from the different indexes. They're identical.